### PR TITLE
Implemented Docker Compose Profiles 

### PR DESCRIPTION
**Pull request by akashvarun10**
solves #70

### CODE REVIEW

**Comment by akashvarun10** [comment]
@amandesai01 added profiles for dev and prod .

**Comment by manasgandhi99** [comment]
@akashvarun10  I just had a question. I could be wrong here as well. The environment variables in the 'app-prod' service are not loaded from a .env file as they are being loaded in 'app' service. Rather they are hardcoded. So is it necessary for the app-prod service or is it not so much important right now?

**Comment by akashvarun10** [comment]
It's not strictly necessary to load environment variables from a .env file for the app-prod service right now; hardcoding them is fine for our current setup. When we move to a CI/CD pipeline or Kubernetes deployment, we can switch to using GitHub Secrets, Kubernetes ConfigMaps/Secrets or other secure methods to manage these variables.

**Comment by akashvarun10** [comment]
@amandesai01 updated profiling for docker compose and also changed to compose.yaml for best practices.

**Review by amandesai01 — CHANGES_REQUESTED** [review]
It should have following behavior:

When running directly (without any specified profile):
It should run in `dev` mode, i.e. should use Docker dev file, mount existing source code volume, etc. It should also spin up database.

When running with profile `prod`:
It should run in `prod` mode, i.e. should use Docker prod file, should NOT spin up database, no volumes should be attached except `static_data`.

IN ALL THE CASES, IT SHOULD BE POSSIBLE TO OVERRIDE ENV VARS, thus, USE SENSIBLE DEFAULTS AND ALLOW OVERRIDE.

**Comment by akashvarun10** [comment]
@amandesai01 

There's no official way to set a default profile in Docker Compose, but I found a tweak around from `StackOverflow` that using an empty string `' '` as a profile works as a workaround. This approach starts services when no profile is specified and prevents them from starting with other profiles, though it may not be future-proof.

and the other workaround I found was setting `COMPOSE_PROFILES` in `/etc/environment` or `~/.bashrc`/`~/.zshrc`  and ensuring consistent profile usage across sessions. However, this method affects all users or requires shell restarts, which might not be ideal for all environments.

Also, it can lead to conflicts if different projects need different profiles, and global settings may not be suitable if you're managing multiple Docker setups.

So, the 1st approach works for now or else we can keep it like before dev and prod ..

**Comment by amandesai01** [comment]
@akashvarun10 got it. so let's have `dev` and `prod` profiles.

**Comment by akashvarun10** [comment]
@amandesai01 I've updated the code and readme..

File: compose.yaml [hunk 1]
```diff
@@ -1,5 +1,3 @@
-version: '3.8'
```
Comment (amandesai01) [inline compose.yaml:1]: why we deleting version?

Comment (akashvarun10) [reply compose.yaml:1]: It's actually deprecated that's why you will be getting warning but it's just a warning.

**Pull request merged**